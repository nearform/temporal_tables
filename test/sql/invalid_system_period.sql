CREATE TABLE invalid_system_period (sys_period bigint);

CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);

INSERT INTO invalid_system_period DEFAULT VALUES;

CREATE TABLE invalid_system_period2 (sys_period tstzrange);

ALTER TABLE invalid_system_period2 DROP COLUMN sys_period;

CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period2
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);

INSERT INTO invalid_system_period2 DEFAULT VALUES;

CREATE TABLE invalid_system_period3 (sys_period tstzrange[]);

CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period3
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);

INSERT INTO invalid_system_period3 DEFAULT VALUES;

CREATE TABLE invalid_system_period4 (sys_period tsrange);

CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period4
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);

INSERT INTO invalid_system_period4 DEFAULT VALUES;