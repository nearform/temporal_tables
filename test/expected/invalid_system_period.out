CREATE TABLE invalid_system_period (sys_period bigint);
CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);
INSERT INTO invalid_system_period DEFAULT VALUES;
ERROR:  system period column "sys_period" of relation "invalid_system_period" is not a range but type bigint
CREATE TABLE invalid_system_period2 (sys_period tstzrange);
ALTER TABLE invalid_system_period2 DROP COLUMN sys_period;
CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period2
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);
INSERT INTO invalid_system_period2 DEFAULT VALUES;
ERROR:  column "sys_period" of relation "invalid_system_period2" does not exist
CREATE TABLE invalid_system_period3 (sys_period tstzrange[]);
CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period3
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);
INSERT INTO invalid_system_period3 DEFAULT VALUES;
ERROR:  system period column "sys_period" of relation "invalid_system_period3" is not a range but an array
CREATE TABLE invalid_system_period4 (sys_period tsrange);
CREATE TRIGGER versioning_trigger
BEFORE INSERT OR UPDATE OR DELETE ON invalid_system_period4
FOR EACH ROW EXECUTE PROCEDURE versioning('sys_period', NULL, false);
INSERT INTO invalid_system_period4 DEFAULT VALUES;
ERROR:  system period column "sys_period" of relation "invalid_system_period4" is not a range of timestamp with timezone but of type timestamp without time zone
